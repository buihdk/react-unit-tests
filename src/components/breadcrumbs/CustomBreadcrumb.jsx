import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Breadcrumb } from 'antd';

const styles = {};

const { Item } = Breadcrumb;

const CustomBreadcrumb = ({ list }) =>
  list && (
    <Breadcrumb separator=">" className={styles.pb15}>
      {list.map(item => (
        <Item key={item}>
          {item.link ? (
            <Link to={item.link}>
              {item.icon} {item.pageName}
            </Link>
          ) : (
            item.pageName
          )}
        </Item>
      ))}
    </Breadcrumb>
  );

CustomBreadcrumb.propTypes = {
  list: PropTypes.arrayOf(PropTypes.object),
};

CustomBreadcrumb.defaultProps = {
  list: [],
};

export default CustomBreadcrumb;
