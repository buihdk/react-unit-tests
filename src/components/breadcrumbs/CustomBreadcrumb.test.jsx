import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import CustomBreadcrumb from './CustomBreadcrumb';

const props = {
  list: [
    {
      pageName: 'Campaign List',
      link: '/',
    },
    {
      pageName: 'Campaign Detail',
    },
  ],
};

describe('Custom Breadcrumb', () => {
  test('renders without crashing ', () => {
    const wrapper = shallow(<CustomBreadcrumb />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test('should render with items', () => {
    const wrapper = shallow(<CustomBreadcrumb {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
