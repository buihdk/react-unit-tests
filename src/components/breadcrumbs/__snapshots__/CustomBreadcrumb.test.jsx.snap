// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Breadcrumb renders without crashing  1`] = `
<Breadcrumb
  separator=">"
/>
`;

exports[`Custom Breadcrumb should render with items 1`] = `
<Breadcrumb
  separator=">"
>
  <BreadcrumbItem
    key="[object Object]"
  >
    <Link
      to="/"
    >
       
      Campaign List
    </Link>
  </BreadcrumbItem>
  <BreadcrumbItem
    key="[object Object]"
  >
    Campaign Detail
  </BreadcrumbItem>
</Breadcrumb>
`;
